tests/ga/test_extension.py

```diff
@@ -1593,6 +1833,534 @@ def test_it_should_parse_boolean_values_encoded_as_strings(self):
+              {
+                "cores": -1,
+                "limit_percentage": 25
+              }
+            ]
+          }
+        }
+      }
+    }
+    '''
+        handler_config = HandlerConfiguration(json.loads(data))
+        resource_config = handler_config.get_resource_configurations()
+
+        self.assertEqual(resource_config.get_cpu_limits_for_extension(), None)
+
+        data = '''{
+      "name": "ExampleHandlerLinux",
+      "version": 1.0,
+      "handlerConfiguration": {
+        "linux": {
+          "resources": {
+            "cpu": [
+              {
+                "cores": 2,
+                "limit_percentage": 251
+              },
+              {
+                "cores": 8,
+                "limit_percentage": 20
+              },
+              {
+                "cores": -1,
+                "limit_percentage": 15
+              }
+            ]
+          }
+        }
+      }
+    }
+            '''
+        handler_config = HandlerConfiguration(json.loads(data))
+        resource_config = handler_config.get_resource_configurations()
+
+        self.assertEqual(resource_config.get_memory_limits_for_extension(), None)
+        self.assertEqual(resource_config.get_cpu_limits_for_extension(), None)
+
+        data = '''{
+      "name": "ExampleHandlerLinux",
+      "version": 1.0,
+      "handlerConfiguration": {
+        "linux": {
+          "resources": {
+            "cpu": [
+              {
+                "cores": 2,
+                "limit_percentage": 0.25
+              },
+              {
+                "cores": 8,
+                "limit_percentage": 20
+              },
+              {
+                "cores": -1,
+                "limit_percentage": 15
+              }
+            ]
+          }
+        }
+      }
+    }
+    '''
+        handler_config = HandlerConfiguration(json.loads(data))
+        resource_config = handler_config.get_resource_configurations()
+
+        self.assertEqual(resource_config.get_memory_limits_for_extension(), None)
+        self.assertEqual(resource_config.get_cpu_limits_for_extension(), None)
+
+        data = '''{
+      "name": "ExampleHandlerLinux",
+      "version": 1.0,
+      "handlerConfiguration": {
+        "linux": {
+          "resources": {
+            "cpu": [
+              {
+                "cores": 2,
+                "limit_percentage": 25
+              },
+              {
+                "cores": 8,
+                "limit_percentage": 20
+              },
+              {
+                "cores": -1,
+                "limit_percentage": "15"
+              }
+            ]
+          }
+        }
+      }
+    }
+    '''
+        handler_config = HandlerConfiguration(json.loads(data))
+        resource_config = handler_config.get_resource_configurations()
+
+        self.assertEqual(resource_config.get_memory_limits_for_extension(), None)
+        self.assertEqual(resource_config.get_cpu_limits_for_extension(), None)
```

Comment:
I think I'm missing why the cpu limits wouldn't be set in this case, can you elaborate?

Reply:
We can sync tomorrow to describe the case. tl;dr- If the json is incorrect or invalid, resource config won't be populated.